pkg/registry/core/serviceaccount/storage/storage_test.go

```diff
@@ -38,13 +49,35 @@ func newStorage(t *testing.T) (*REST, *etcd3testing.EtcdTestServer) {
 		DeleteCollectionWorkers: 1,
 		ResourcePrefix:          "serviceaccounts",
 	}
-	rest, err := NewREST(restOptions, nil, nil, 0, nil, nil, nil, false)
+	// set issuer, podStore and secretStore to allow the token endpoint to be initialised
+	rest, err := NewREST(restOptions, fakeTokenGenerator{"fake"}, nil, 0, panicGetter{}, panicGetter{}, nil, false)
 	if err != nil {
 		t.Fatalf("unexpected error from REST storage: %v", err)
 	}
 	return rest, server
 }
 
+// A basic fake token generator which always returns a static string
+type fakeTokenGenerator struct {
+	staticToken string
+}
+
+func (f fakeTokenGenerator) GenerateToken(claims *jwt.Claims, privateClaims interface{}) (string, error) {
+	return f.staticToken, nil
+}
+
+var _ token.TokenGenerator = fakeTokenGenerator{}
+
+// Currently this getter only panics as the only test case doesn't actually need the getters to function.
+// When more test cases are added, this getter will need extending/replacing to have a real test implementation.
```

Comment:
Whilst it would be nice to add additional tests now, this whole area is still covered by integration tests so we aren't running totally blind :)